kork-plugins/src/main/java/com/netflix/spinnaker/kork/plugins/spring/PluginLoader.java

```diff
@@ -129,23 +128,31 @@ private boolean checkFileExists() {
    * @return List<URL> List of paths to jars, where enabled is true
    */
   private URL[] getJarPathsFromPluginConfigurations(
-      List<PluginProperties.PluginConfiguration> pluginConfigurations) {
+      List<PluginProperties.PluginConfiguration> pluginConfigurations, boolean downloadingEnabled) {
     return pluginConfigurations.stream()
         .map(PluginProperties.PluginConfiguration::getJars)
         .flatMap(Collection::stream)
-        .map(Paths::get)
-        .map(this::getUrlFromPath)
+        .map(s -> convertToUrl(s, downloadingEnabled))
         .distinct()
         .toArray(URL[]::new);
   }
 
   /**
-   * @param path
+   * @param jarLocation
    * @return path as a URL
    */
-  private URL getUrlFromPath(Path path) {
+  private URL convertToUrl(String jarLocation, boolean downloadingEnabled) {
     try {
-      return path.toUri().toURL();
+      if (jarLocation.startsWith("/")) {
+        return Paths.get(jarLocation).toUri().toURL();
+      } else {
+        if (downloadingEnabled) {
```

Comment:
Technically `jarLocation` could also be "file://" which is not downloading. You can move your test a bit and check on the URL's protocol.